From ddfd6e05dfae0c56c835ec584581b561809a2502 Mon Sep 17 00:00:00 2001
From: Pete Batard <pete@akeo.ie>
Date: Mon, 27 Jan 2025 19:25:14 +0000
Subject: [PATCH] OpenSSL submodule fixes for ARM compilation

* Double randomness in drbg_add() was NEVER used in the first place.
---
 crypto/rand/rand_meth.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crypto/rand/rand_meth.c b/crypto/rand/rand_meth.c
index 276763057d..74a8892127 100644
--- a/crypto/rand/rand_meth.c
+++ b/crypto/rand/rand_meth.c
@@ -25,7 +25,7 @@ static int drbg_add(const void *buf, int num, double randomness)
 /* Implements the default OpenSSL RAND_seed() method */
 static int drbg_seed(const void *buf, int num)
 {
-    return drbg_add(buf, num, num);
+    return drbg_add(buf, num, 0);
 }
 
 /* Implements the default OpenSSL RAND_status() method */
-- 
2.45.2.windows.1

